mat-process-api/V1/Validators/GetProcessImageRequestValidator.cs

```diff
@@ -0,0 +1,27 @@
+using FluentValidation;
+using mat_process_api.V1.Boundary;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace mat_process_api.V1.Validators
+{
+    public class GetProcessImageRequestValidator : AbstractValidator<GetProcessImageRequest>, IGetProcessImageRequestValidator
+    {
+        public GetProcessImageRequestValidator()
+        {
+            ValidatorOptions.CascadeMode = CascadeMode.StopOnFirstFailure; // Stops the validator from going into DependentRules or deeper into any single rule in general. This is needed because it doesn't make sense to keep checking if property is empty, once it was determined it's null.
+
+            RuleFor(req => req.processRef).NotNull().WithMessage("Process reference must be provided.").NotEmpty().WithMessage("Process reference must be provided.").Must(ValidateGuid)
+               .WithMessage("You need to provide a valid process reference.");
+            RuleFor(req => req.imageId).NotNull().WithMessage("Image Id must be provided.").NotEmpty().WithMessage("Image Id must be provided.").Must(ValidateGuid)
+                .WithMessage("You need to provide a valid Image Id.");
```

Comment:
The request validator is lacking 2 Rules for making 2 newly added GetProcessImageRequest properties required. (processType, fileExtension)

Reply:
Actually, the rules will only be required to check that the image ID and process ref are GUIDs. This is because the rest of the variables are now part of the "path" and if you do not supply them, the API throws 404.

Reply:
Resolved. After a bit of testing it was discovered that it's possible to pass a whitespace (like '%20') into the endpoint call and it would be interpreted by ASP.NET as 'null' value after doing the model binding. It was decided by the team to add the missing validation for 'NotNull' 'NotEmpty' to have more distinct error messages.